### ISSUE TITLE
setImmediate should return timerId

### ISSUE THREAD

**Issue by JiaLiPassion** [jvilk/BrowserFS#169]
the code here [link](https://github.com/jvilk/BrowserFS/blob/9f3d0501fdc0b7b9e94406320b7fbb787ad7dbf5/src/index.ts#L41), browserFS patched window.setImmediate but without return a timerId, so it will cause some errors if other library use the timerId, for example, this issue https://github.com/angular/zone.js/issues/560.

**Comment by sstone1** [comment]
I raised the original issue against zone.js - I can confirm that removing BrowserFS from my configuration fixes the problem, as does using an alternative setImmediate implementation (I used https://github.com/YuzuJS/setImmediate).

I had to use the following configuration for webpack to force it to load setImmediate before BrowserFS:

```javascript
module.exports = function (options) {
  return {
    module: {
      rules: [
        /**
         * https://github.com/jvilk/BrowserFS/issues/169
         */
        {
          test: /browserfs.*\.js$/,
          loader: 'imports-loader?importedSetImmediate=setimmediate'
        }
      ],
    }
  };
}
```

I needed this because it looks like BrowserFS's version of the process module gets imported very early on by webpack or Angular 2.

**Comment by jvilk** [comment]
Sorry about that. The `setImmediate` global is a remnant from an earlier time; BrowserFS should probably not globally define it at all!

It's non-standard for speed (avoids storing arguments, doesn't support cancellation, etc), which is great for BrowserFS and terrible for all of you who want to use it.

I will remove it as a global in the next major release, since BFS customers may depend on it. Thanks for posting your workarounds.

**Comment by jvilk** [comment]
Fixed in v1.3.0.
